src/OrchardCore.Modules/OrchardCore.Queries/Sql/SqlQuerySource.cs

```diff
@@ -48,51 +48,40 @@ public async Task<IQueryResults> ExecuteQueryAsync(Query query, IDictionary<stri
             var tokenizedQuery = await _liquidTemplateManager.RenderStringAsync(sqlQuery.Template, NullEncoder.Default,
                 parameters.Select(x => new KeyValuePair<string, FluidValue>(x.Key, FluidValue.Create(x.Value, _templateOptions))));
 
-            var connection = _dbConnectionAccessor.CreateConnection();
             var dialect = _session.Store.Configuration.SqlDialect;
 
             if (!SqlParser.TryParse(tokenizedQuery, _session.Store.Configuration.Schema, dialect, _session.Store.Configuration.TablePrefix, parameters, out var rawQuery, out var messages))
             {
                 sqlQueryResults.Items = Array.Empty<object>();
-                connection.Dispose();
+
                 return sqlQueryResults;
             }
 
-            if (sqlQuery.ReturnDocuments)
-            {
-                IEnumerable<long> documentIds;
-                using (connection)
-                {
-                    await connection.OpenAsync();
-
-                    using var transaction = connection.BeginTransaction(_session.Store.Configuration.IsolationLevel);
-                    documentIds = await connection.QueryAsync<long>(rawQuery, parameters, transaction);
-                }
+            using var connection = _dbConnectionAccessor.CreateConnection();
+            await connection.OpenAsync();
+            using var transaction = connection.BeginTransaction(_session.Store.Configuration.IsolationLevel);
```

Comment:
Connection and Transaction are `IAsyncDisposable` so we can even use `await using var ...`

Reply:
We don't use await for others connections, is it better to change it everywhere we create a connection?

Reply:
We now use it for the shell scope that we made async disposable, so that we call `Session.DisposeAsync()` that call `_transaction.DisposeAsync()` and `_connection.DisposeAsync()`. So yes, not sure it has a big impact but at least to be consistent.

For info I noticed that when the YesSql session is disposed the connection is first closed, but I assume that disposing it also close it but I didn't check it.

            if (_connection != null)
            {
                await _connection.CloseAsync();
                await _connection.DisposeAsync();
                _connection = null;
            }

Reply:
I am guessing disposing will close the connection. But maybe it's better to explicitly close the connection. I'll add closeAsync everywhere ti create it

Reply:
Worth to check it because it can't be part of the inner try/finally of the using block.

Reply:
Why can't it be part of a finally block?

Reply:
I'm talking about the inner finally block done for us when we use `[await] using var ...`.

So your own finally block may run after and will try to close an already disposed instance

We would need to not use an using block but only our try/finally blocks

Reply:
Just looked at the `SqlConnection` implementation, it doesn't seem to have a `DisposeAsync()` but looks like `Close()` do additionnal stuff.

Reply:
@jtkech I added `CloseAsync()`